### PR TITLE
Debugger: Modernize and increase timeout

### LINKED ISSUE #12470: Debugger / Site Health: ensure results match Jetpack.com Debug
On some sites, when going to Tools > Site Health, you may get failing tests with a timeout error:

image

Running tests on the Jetpack.com Debugger, however, does not return any issues as we do not run into the same timeout there.

It would be nice if the 2 tools reported matching results at all times.

**********

Originally reported in 3937-gh-jpop-issues

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes #12470

Adds a 30-second timeout to the other connection test method in the debug suite, pending actually getting that down to only one test.

Also removes pre-WP 5.2 era and pre-PHP 5.3 things that are no longer needed as soon as WP 5.3 ships (which will be before 8.0, so 8.0 will be bumping things to 5.2).

If we need to hold this until 5.2 as our minimum is official, that's fine.

#### Changes proposed in this Pull Request:
* Debugger: Increase timeout for an occasionally slow test and modernize code.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* n/a

#### Testing instructions:
* Visit the Debug area via the footer of a Jetpack settings page. Any issues or errors?
* Run the Site Health tool. Try on a slow site.

#### Proposed changelog entry for your changes:
* Debugger: Increase timeout for an occasionally slow test.

### CODE REVIEW

**Comment by kraftbj** [comment]
As an aside, I'd like to work this into a package—would be great as a dependent on the connection package, but want to get this improvement in without having it be blocked by packaging.

File: _inc/lib/debugger/class-jetpack-debugger.php [hunk 1]
```diff
@@ -5,56 +5,31 @@
...
  */
 class Jetpack_Debugger {
-
-	/**
-	 * Determine the active plan and normalize it for the debugger results.
-	 *
-	 * @return string The plan slug prepended with "JetpackPlan"
-	 */
-	private static function what_jetpack_plan() {
-		// Specifically not deprecating this function since it modifies the output of the Jetpack_Debug_Data::what_jetpack_plan return.
```
Comment (jeherve) [inline _inc/lib/debugger/class-jetpack-debugger.php]: Is this okay to remove this now? The comment makes me worry. :)

**Review by jeherve — APPROVED** [review]
This tests well on my end. 👍 I only have one question.

File: _inc/lib/debugger/class-jetpack-debugger.php (hunk 1 above)
Comment (kraftbj) [reply _inc/lib/debugger/class-jetpack-debugger.php]: Yeah, that function provided some of the "debug data" that was part of the appended data when using the contact form that was built into Jetpack and all that debug data was legacy for WP versions prior to 5.2 (the old "Advanced Debug Results" link. The specific format of the text was required for Kayako/Zendesk processing to assign a plan value to a ticket.

For WP 5.2+, this stuff never displayed anywhere and with all private support requests going to the contact form (which uses the API and doesn't parse text anyhow), the plan output isn't needed here nor the legacy format that we needed to maintain.

**Pull request merged**